processor/transformprocessor/config.go

```diff
@@ -41,13 +41,18 @@ type Config struct {
 	LogStatements    []common.ContextStatements `mapstructure:"log_statements"`
 
 	FlattenData bool `mapstructure:"flatten_data"`
+	logger      *zap.Logger
 }
 
 var _ component.Config = (*Config)(nil)
 
 func (c *Config) Validate() error {
 	var errors error
 
+	if c.logger != nil && metrics.UseConvertBetweenSumAndGaugeMetricContext.IsEnabled() {
+		c.logger.Sugar().Infof("Metric conversion functions use metric context since %s is enabled", metrics.UseConvertBetweenSumAndGaugeMetricContext.ID())
```

Comment:
```suggestion
		c.logger.Sugar().Infof("Metric conversion functions use metric context since %s is enabled. If your statements are not parsing, check if you're using the metrics conversion functions via the datapoint context.", metrics.UseConvertBetweenSumAndGaugeMetricContext.ID())
```